plugins/quickicon/eos/src/Extension/Eos.php

```diff
@@ -0,0 +1,382 @@
+     *
+     * @param   QuickIconsEvent  $event  The event object
+     *
+     * @return  void
+     *
+     * @since __DEPLOY_VERSION__
+     *
+     * @throws Exception
+     */
+    public function getEndOfServiceNotification(QuickIconsEvent $event)
+    {
+        $context = $event->getContext();
+
+        if ($context !== $this->params->get('context', 'update_quickicon')) {
+            return;
+        }
+
+        if (!$this->shouldDisplayMessage()) {
+            return;
+        }
+
+        // No messages yet
+        if (!$this->currentMessage) {
+            return;
+        }
+
+        // Show this only when not snoozed
+        if ($this->params->get('last_snoozed_id', 0) < $this->currentMessage['id']) {
+            // Build the  message to be displayed in the cpanel
+            $messageText = Text::sprintf($this->currentMessage['messageText'], HTMLHelper::_('date', Eos::EOS_DATE, Text::_('DATE_FORMAT_LC3')), $this->currentMessage['messageLink']);
+            if ($this->currentMessage['snoozable']) {
+                $messageText .= '<p><button class="btn btn-warning eosnotify-snooze-btn" type="button" >' . Text::_('PLG_QUICKICON_EOS_SNOOZE_BUTTON') . '</button></p>';
+            }
+            $this->getApplication()->enqueueMessage($messageText, $this->currentMessage['messageType']);
+        }
+        try {
+            $this->document->getWebAssetManager()->registerAndUseScript('plg_quickicon_eos.script', 'plg_quickicon_eos/snooze.js', [], ['type' => 'module']);
+        } catch (Exception $e) {
+            echo $e->getMessage();
+            exit();
+        }
+        // The message as quickicon
+        // Add the icon to the result array
+        $result               = $event->getArgument('result', []);
+        $messageTextQuickIcon = Text::sprintf($this->currentMessage['quickiconText'], HTMLHelper::_('date', Eos::EOS_DATE, Text::_('DATE_FORMAT_LC3')));
+
+        // The message as quickicon
+
+        $result[] = [
+            [
+                'link'  => $this->currentMessage['messageLink'],
+                'image' => 'fa fa-life-ring',
+                'text'  => $messageTextQuickIcon,
+                'id'    => 'plg_quickicon_eos',
+                'group' => $this->currentMessage['groupText'],
+                'class' => 'success',
+            ],
+        ];
+
+        $event->setArgument('result', $result);
+    }
+
+    /**
+     * Save the plugin parameters
+     *
+     * @return  bool
+     *
+     * @since __DEPLOY_VERSION__
+     */
+    private function saveParams(): bool
+    {
+        $db    = Factory::getContainer()->get('DatabaseDriver');
+        $query = $db->getQuery(true)->update($db->quoteName('#__extensions'))->set($db->quoteName('params') . ' = ' . $db->quote($this->params->toString('JSON')))->where($db->quoteName('type') . ' = ' . $db->quote('plugin'))->where($db->quoteName('folder') . ' = ' . $db->quote('quickicon'))->where($db->quoteName('element') . ' = ' . $db->quote('eos'));
```

Comment:
```suggestion
        $params = $this->params->toString('JSON');
        $db     = $this->getDatabase();
        $query  = $db->getQuery(true)
            ->update($db->quoteName('#__extensions'))
            ->set($db->quoteName('params') . ' = :params')
            ->where($db->quoteName('type') . ' = ' . $db->quote('plugin'))
            ->where($db->quoteName('folder') . ' = ' . $db->quote('quickicon'))
            ->where($db->quoteName('element') . ' = ' . $db->quote('eos'))
            ->bind(':params', $params);
```
use `$this->getDatabase()`
use `->bind()`
code style (line length)